File: Online_Assessment_portal_backend/src/main/java/com/test/entity/Result.java
package com.test.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "OAP_result")
public class Result {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	
	
	@Column(name = "result_status")
	private String status;

	@Column(name = "result_score")
	private String score;

	@Column(name = "exam_date")
	private String examDate;

	@Column(name = "total_marks")
	private String totalMarks;

	@Column(name = "total_question")
	private String totalQuestion;

	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "exam_name")
	private Subject subName;
	
	@ManyToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "username")
	private User username;

	@ManyToOne
	private Exam examId;

}

File: Online_Assessment_portal_Frontend/src/components/AdminComponent/AdminDashboard/ResultComponent/Result.js

import { useEffect } from "react";
import style from "../SubjectComponent/Subject.module.css"
import { getAllStudentsResult, getStudentsResultByUserName } from "../../../../store/slices/ResultSlice";
import { useDispatch, useSelector } from "react-redux";

function Result() {


    const dispatch = useDispatch();

    //----------------to get All Students Results-----------------------


    const { post, isLoading } = useSelector((state) => ({ ...state.results }));

    useEffect(() => {
        dispatch(getAllStudentsResult())
    }, [])

    if (isLoading) {
        return <div>Loading results...</div>
    }

    console.log(post.subName);

    return (
        <>
            <div id={style.displayHeadingBox}>
                <h2>Exam List</h2>
            </div>

            <div id={style.tableBox}>
                <table>
                    <thead>
                        <tr>
                            <th id="center">User email</th>
                            <th id="center">Exam Name</th>
                            <th id="center">Exam Date</th>
                            <th id="center">Result Status</th>
                            <th id="center">Your Score</th>
                            <th id="center">Total Marks</th>
                            <th id="center">Total Question</th>
                        </tr>
                    </thead>
                    <tbody>
                        {
                            post.map((data, i) => {
                                return (
                                    <tr key={i}>
                                        <td>{data.username.username}</td>
                                        <td>{data.subName}</td>
                                        <td>{data.examDate}</td>
                                        <td>{data.status}</td>
                                        <td>{data.score}</td>
                                        <td>{data.totalMarks}</td>
                                        <td>{data.totalQuestion}</td>
                                    </tr>
                                );
                            })
                        }

                    </tbody>
                </table>
            </div>

        </>
    );
}

export default Result;